src/Views/IMEView.cpp

```diff
@@ -0,0 +1,162 @@
+	log_printf(DBG_DEBUG, "Created IMEView \"%s\"", title);
+	utf8_to_utf16((uint8_t *)title, _title);
+	utf8_to_utf16((uint8_t *)showText, _showText);
+	_result = result;
+	if (initialText) {
+		log_printf(DBG_DEBUG, "initialText set \"%s\"", initialText);
+		utf8_to_utf16((uint8_t *) initialText, _initialText);
+	} else {
+		*_initialText = '\0';
+	}
+	_maxTextLength = maxInputLength;
+}
+
+IMEView::~IMEView() {
+	sceImeDialogTerm();
+	delete _input_text_buffer_utf16;
+	delete _input_text_buffer_utf8;
+}
+
+
+int IMEView::Display() {
+	if (request_destroy) // done here!
+		return 0;
+
+	if (!shown_dialog) {
+		log_printf(DBG_DEBUG, "Initializing IMEView");
+		_input_text_buffer_utf16 = new uint16_t[_maxTextLength + 1];
+		_input_text_buffer_utf8 = new uint8_t[_maxTextLength + 1];
+
+		SceImeDialogParam param;
+		sceImeDialogParamInit(&param);
+
+		param.supportedLanguages = 0x0001FFFF;
+		param.languagesForced = SCE_TRUE;
+//		param.type = SCE_IME_TYPE_BASIC_LATIN;
+		param.title = _title;
+		param.enterLabel = SCE_IME_ENTER_LABEL_SEARCH;
+//		param.dialogMode = SCE_IME_DIALOG_DIALOG_MODE_WITH_CANCEL;
+		param.maxTextLength = _maxTextLength;
+		param.inputMethod = 0;
+		if (_initialText && *_initialText != '\0') {
+			param.initialText = _initialText;
+		}
+		param.inputTextBuffer = _input_text_buffer_utf16;
+
+		SceInt32 res = sceImeDialogInit(&param);
+		shown_dialog = res == 0;
+		if (!shown_dialog) {
+			log_printf(DBG_ERROR,
+			           "Couldn't open IMEView dialog: 0x%lx\nFind the status codes at "
+			           "https://github.com/vitasdk/vita-headers/blob/master/include/psp2/common_dialog.h",
+			           res);
+			if (_result)
+				_result->status = IMEVIEW_STATUS_CANCELED;
+			request_destroy = true;
+		}
+		return 0;
+	}
+
+	auto new_status = (IMEViewStatus)sceImeDialogGetStatus();
+	if (_status != new_status)
+		switch (new_status) {
+			case IMEVIEW_STATUS_NONE:
+				log_printf(DBG_DEBUG, "IMEView status \"IMEVIEW_STATUS_NONE\"");
+				break;
+			case IMEVIEW_STATUS_RUNNING:
+				log_printf(DBG_DEBUG, "IMEView status \"IMEVIEW_STATUS_RUNNING\"");
+				break;
+			case IMEVIEW_STATUS_FINISHED:
+				log_printf(DBG_DEBUG, "IMEView status \"IMEVIEW_STATUS_FINISHED\"");
+				break;
+			case IMEVIEW_STATUS_CANCELED:
+				log_printf(DBG_DEBUG, "IMEView status \"IMEVIEW_STATUS_CANCELED\"");
+				break;
+		}
+	_status = new_status;
+
+	if (_status == IMEVIEW_STATUS_FINISHED) {
+		SceImeDialogResult result={};
+		sceImeDialogGetResult(&result);
+
+		if (result.button == SCE_IME_DIALOG_BUTTON_CLOSE)
+			_status = IMEVIEW_STATUS_CANCELED;
+		else {
+			utf16_to_utf8(_input_text_buffer_utf16, _input_text_buffer_utf8);
+			if (_result)
+				_result->userText = std::string((char *)_input_text_buffer_utf8);
+		}
+
+		sceImeDialogTerm();
+		request_destroy = true;
+	}
+
+	if (_result)
+		_result->status = _status;
+
+	return 0;
+}
+
+// copied from https://github.com/devingDev/VitaCord/blob/master/src/VitaIME.cpp
```

Comment:
Not their code.

Reply:
Might be but I've copied it from there. Should I remove the comment and replace it with something else? Should I remove it without replacement?

Reply:
I think you can credit Vitashell 👍

Reply:
Well, I'm now using `std::wstring_convert` so I got rid of all of that code and the credit problem as well ;)